Add k6 studio docs initial content (#1783)
grafana/k6-docs: docs/sources/k6-studio/components/test-generator.md

Reviewer: This will need more feedback, if there was an image it was no loading for me :(

2. **Test Generator actions**: On the top-right you can see the action buttons for the Test Recorder. From here you can click **Save Generator** to save changes to your test generator file, or click the menu icon to:
   - **Validate script**: Opens the [Test Validator](https://grafana.com/docs/k6-studio/components/test-validator/) and starts a one iteration run of the test script.
   - **Export script**: Opens the export script dialog box. You can enter a name for your script, and also select whether you want to overwrite a script if one with the same name already exists.
   - **Delete generator**: Deletes the selected test generator.
3. **Test Generator options**: Below the test recording name, you can see:
   - **Requests**: The total number of requests in the recording
   - **Show static assets**: A toggle that controls whether you can see all static assets requests in the Requests list. The static assets requests are hidden by default.
   - **Filter**: A search box that allows you to filter the list of requests by URL, method (such as GET or POST), and status code.

> 4. **Test rules list**: The list of requests, and groups if any, in the HAR file. The requests are organized by time, and you can see the method, status code, host, and path for each one. You can also collapse and expand groups to inspect them more easily.

Reviewer: this might be wrong, will need to see the image to properly assess it 🤔

## Test options

The test options tab allows you to configure three separate parts of your test script:

- Load profile
- Think time
- Test data

### Load profile

The load profile controls how k6 schedules VUs and iterations to run your performance test.

There are two executors available under load profiles:

- **Ramping VUs**: A variable number of VUs execute as many iterations as possible for a specified amount of time.
- **Shared iterations**: A fixed number of iterations are shared between a number of VUs, and the test ends once all iterations have been executed.

Each executor has different variables you can configure. For more details, refer to [Executors](https://grafana.com/docs/k6/latest/using-k6/scenarios/executors/).

### Think time

The think time option allows you to configure a fixed or random delay, between groups or iterations, to simulate a more realistic test. This isn't required, but it's a best practice when creating and running performance tests.

### Test data

> The test data option allows you to define variables, which you can then use in your [custom code rules](#custom-code-rule).

Reviewer: This can also be used in the `Parameterization rule`

After you define a variable, you can refer to them in your custom code rules by using: `VARS["VARIABLE_NAME"]`.

## Allowed hosts

The **Allowed hosts** option lets you configure which hosts you want to include in your test script. It shows all the hosts that are identified from your HAR recording file, and you can choose to select which ones you want to include or remove from your test script.

For performance testing, it's common to not include static assets, or remove hosts from 3rd-party services from your tests, but you can include them depending on your use case.

## Rules

Test rules are rules you can add and configure to your test generator, that allow you to customize the generated test script. These rules can use the information from a test recording, to then generate code changes to help make your test scripts more reliable, and reusable.

The available rules are:

- Verification rule
- Correlation rule

> - Custom code rule

Reviewer: We also have the `Parameterization rule` now :)

You can add multiple correlation and custom code rules to your test generator.

### Verification rule

The verification rule is a default rule that's created for every test generator. It adds a [check](https://grafana.com/docs/k6/latest/using-k6/checks/) statement after every request that validates if the response status code is the same as the one from the test recording.

The verification rule is useful for making sure that your systems are performing as expected. Failed checks do not cause a test to abort, or interrupt a test run. Checks generate a metric that you can inspect after a test run is completed to understand how your system performed.

For more details about checks, refer to [Checks](https://grafana.com/docs/k6/latest/using-k6/checks/).

### Correlation rule

> The correlation rule allows you to extract data from your test recording, and reuse it across your test script. This is a common use case when working with applications that have unique IDs that are generated for a request, and then are passed to every subsequent request, for example.

Reviewer: ```suggestion
The correlation rule allows you to extract dynamic data at runtime, and reuse it across your test script. This is a common use case when working with applications that have unique IDs that are generated for a request, and then are passed to every subsequent request, for example.
```

The correlation rule includes an Extractor and a Replacer. They both have the same configuration options, but the Extractor can be used to find and extract the value you want to use across the script, and the Replacer can be used to find the places where you want to replace and use the extracted value.

The configuration fields are:

- **Filter**: Define a request path that this filter applies to. Plain text and regular expression are supported.
- **Target**: Select the headers, body, or URL path as the target for the extractor or replacer.

> - **Type**: Select Begin-End or Regex as the way to search for the value to be extracted or replaced.

Reviewer: we also support `json` as type
It allows you to define a json key to retrieve the value from the body. 
ex.
```
title
[0].title
```

  - **Begin-End**: Define the Begin and End values as the strings immediately before and after the value to be extracted or replaced.
  - **Regex**: Define the regular expression to match the value to be extracted or replaced.

> When creating or editing a custom rule, you can use the **Rule preview** panel to check that your configuration options are working as intended, and being applied to the correct requests and values in your test script.

Reviewer: ```suggestion
When creating or editing a correlation rule, you can use the **Rule preview** panel to check that your configuration options are working as intended, and being applied to the correct requests and values in your test script.
```
